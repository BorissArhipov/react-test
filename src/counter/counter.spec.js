import React from 'react';
import Counter from './counter';

describe("Counter component", () => {
    it("should render counter component", () => {
        const component = shallow(<Counter/>);
        expect(component).toMatchSnapshot();
    });

    describe("Counter handlers", () => {
        it("should increase the count value by 1", () => {
            const component = shallow(<Counter/>);
            const btn = component.find(".plusOneBtn");
            btn.simulate("click");
            expect(component.state().count).toBe(1);
        });

        it("should reset the count value to 10", () => {
            const component = shallow(<Counter/>);
            const btn = component.find(".resetBtn");
            btn.simulate("click");
            expect(component.state().count).toBe(10);
        });
    });
});
  