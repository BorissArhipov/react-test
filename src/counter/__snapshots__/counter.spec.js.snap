// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component Counter handlers should change count value  1`] = `
<div>
  <div>
    0
  </div>
  <button
    className="plusOneBtn"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="resetBtn"
    onClick={[Function]}
  >
    Reset to 10
  </button>
</div>
`;

exports[`Counter component Counter handlers should increase the count value by 1 1`] = `
<div>
  <div>
    0
  </div>
  <button
    className="plusOneBtn"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="resetBtn"
    onClick={[Function]}
  >
    Reset to 10
  </button>
</div>
`;

exports[`Counter component should render counter component 1`] = `
<div>
  <div>
    0
  </div>
  <button
    className="plusOneBtn"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="resetBtn"
    onClick={[Function]}
  >
    Reset to 10
  </button>
</div>
`;
